java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java

```diff
@@ -55,7 +55,7 @@
                         List<Type> parametersOfReturnType = parseTypeParameters(classSignature.getReturnType().toString(), typeSolver, invokationContext);
                         Type newReturnType = methodUsage.returnType();
                         // consume one parametersOfReturnType at the time
-                        if (!(newReturnType instanceof VoidType)) {
+                        if (newReturnType instanceof ReferenceType && parametersOfReturnType.size() > 0) {
```

Comment:
Maybe you could use `isReferenceType()`

Reply:
Is that for performance reasons?

Reply:
also, but to me it would look nicer. Just a preference, not a big deal